### ISSUE TITLE
encryption key runtime panic

### ISSUE THREAD

**Issue by VVX7** [preludeorg/operator-support#49]
Using multiple encryption keys with TCP reverse shell sometimes causes a runtime panic on Operator 1.10+.  
The agent connects to Operator but panics after receiving a command reverse shell command.

I've double checked that both keys are valid.

Steps to reproduce:
1. Add 2 encryption keys to Operator.
2. Build and run Pneuma using TCP contact.
3. Send any command via reverse shell.

```
2021/01/07 20:40:52 [tcp] agent at PID 147048 using key 
panic: runtime error: makeslice: len out of range
```

**Comment by khyberspache** [comment]
Just saw this - I'll take a look first thing tomorrow morning.

**Comment by khyberspache** [comment]
I haven't been able to replicate this in my setup using the repro instructions.  Are you running a Darwin or Linux agent?

My Keys
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/43142871/104081726-7c118c80-51fe-11eb-82b9-dc7381a601c8.png">

New agent beacons in
<img width="879" alt="image" src="https://user-images.githubusercontent.com/43142871/104081732-8895e500-51fe-11eb-92c7-8a819b1e0027.png">

commands run
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/43142871/104081753-b3803900-51fe-11eb-9003-4f8058d04ef4.png">

**Comment by VVX7** [comment]
Linux agent on a Ubuntu 20.04 host. Can reproduce this on 1.10 and 1.11 AppImage.  Have not tested deb packages.

![image](https://user-images.githubusercontent.com/46228229/104082252-a9f8d000-5202-11eb-995e-b6ee4e5a566a.png)
![image](https://user-images.githubusercontent.com/46228229/104082266-c09f2700-5202-11eb-84ec-00ab784a76f0.png)
![image](https://user-images.githubusercontent.com/46228229/104082278-dc0a3200-5202-11eb-9111-56bcbd4c900b.png)
![image](https://user-images.githubusercontent.com/46228229/104082300-022fd200-5203-11eb-981f-e8b7e5cbf9b5.png)

**Comment by khyberspache** [comment]
Thanks for the details, I'll try to get a reproduction in my Kali environment.

**Comment by khyberspache** [comment]
I just did a series of tests on Kali (2020.3) and I wasn't able to reproduce the issue. Might be an ubuntu thing. I'll download ubuntu and try with that.

![image](https://user-images.githubusercontent.com/43142871/104226143-54e0d800-5415-11eb-9c90-402013bf6fb3.png)

![image](https://user-images.githubusercontent.com/43142871/104226162-5dd1a980-5415-11eb-976b-a9e1c3e5c666.png)

![image](https://user-images.githubusercontent.com/43142871/104226233-7b067800-5415-11eb-824a-49811f14d546.png)

**Comment by khyberspache** [comment]
Figured out how to reproduce - whichever key is listed last will segfault the pneuma agent.

**Comment by khyberspache** [comment]
Fixed in 0.9.12
